File: leetcode/128. Longest Consecutive Sequence/solution.py
# 128. Longest Consecutive Sequence
#
# Solution: walk each streak

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        m = 0
        for x in s:
            if x - 1 not in s:
                y = x + 1
                while y in s:
                    y += 1
                m = max(m, y - x)
        return m

File: leetcode/128. Longest Consecutive Sequence/solution.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s(nums.begin(), nums.end());
        int x, y, i;
        int b = 0;
        for (i = 0; i < nums.size(); i++) {
            x = nums[i];
            if (s.find(x - 1) == s.end()) {
                y = x + 1;
                while (s.find(y) != s.end()) y++;
                b = max(b, y - x);
            }   
        }
        return b;
    }
};